### PR TITLE
fix ambiguity warnings trunc(BigInt,::Float16)

### PR DESCRIPTION
**Pull request by simonbyrne**
cf #13376

### CODE REVIEW

File: base/gmp.jl [hunk 1]
```diff
@@ -121,18 +121,18 @@ end
 convert(::Type{BigInt}, x::Bool) = BigInt(UInt(x))
 
 
-function unsafe_trunc(::Type{BigInt}, x::CdoubleMax)
+function unsafe_trunc(::Type{BigInt}, x::Union(Float32,Float64))
```
Comment (tkelman) [inline base/gmp.jl]: `Union{Float32,Float64}`

Comment (simonbyrne) [inline base/gmp.jl]: Clearly I haven't been making enough PRs lately. Thanks

**Comment by simonbyrne** [comment]
I don't think the remaining ambiguity warnings are my fault.

**Comment by tkelman** [comment]
The remaining ones look like #13397, cc @andreasnoack. This PR lgtm.

**Pull request merged**